spec.html

```diff
@@ -6475,60 +6517,84 @@
+              1. Let _valueDesc_ be the PropertyDescriptor{[[Value]]: _V_}.
+              1. Return ? _Receiver_.[[DefineOwnProperty]](_P_, _valueDesc_).
+            1. Else _Receiver_ does not currently have a property _P_,
+              1. Return ? CreateDataProperty(_Receiver_, _P_, _V_).
+          1. Assert: IsAccessorDescriptor(_ownDesc_) is *true*.
+          1. Let _setter_ be _ownDesc_.[[Set]].
+          1. If _setter_ is *undefined*, return *false*.
+          1. Perform ? Call(_setter_, _Receiver_, &laquo; _V_ &raquo;).
+          1. Return *true*.
+        </emu-alg>
+      </emu-clause>
     </emu-clause>
 
     <!-- es6num="9.1.10" -->
     <emu-clause id="sec-ordinary-object-internal-methods-and-internal-slots-delete-p">
       <h1>[[Delete]] (_P_)</h1>
-      <p>When the [[Delete]] internal method of _O_ is called with property key _P_ the following steps are taken:</p>
+      <p>When the [[Delete]] internal method of _O_ is called with property key _P_, the following steps are taken:</p>
       <emu-alg>
-        1. Assert: IsPropertyKey(_P_) is *true*.
-        1. Let _desc_ be ? _O_.[[GetOwnProperty]](_P_).
-        1. If _desc_ is *undefined*, return *true*.
-        1. If _desc_.[[Configurable]] is *true*, then
-          1. Remove the own property with name _P_ from _O_.
-          1. Return *true*.
-        1. Return *false*.
+        1. Return ? OrdinaryDelete(_O_, _P_).
       </emu-alg>
+
+      <emu-clause id="sec-ordinarydelete" aoid="OrdinaryDelete">
+        <h1>OrdinaryDelete (_O_, _P_)</h1>
+        <p>When the abstract operation OrdinaryDelete is called with Object _O_ and property key _P_, the following steps are taken:</p>
+        <emu-alg>
+          1. Assert: IsPropertyKey(_P_) is *true*.
+          1. Let _desc_ be ? _O_.[[GetOwnProperty]](_P_).
+          1. If _desc_ is *undefined*, return *true*.
+          1. If _desc_.[[Configurable]] is *true*, then
+            1. Remove the own property with name _P_ from _O_.
+            1. Return *true*.
+          1. Return *false*.
+        </emu-alg>
+      </emu-clause>
     </emu-clause>
 
     <!-- es6num="9.1.11" -->
     <emu-clause id="sec-ordinary-object-internal-methods-and-internal-slots-enumerate">
       <h1>[[Enumerate]] ()</h1>
-      <p>When the [[Enumerate]] internal method of _O_ is called the following steps are taken:</p>
+      <p>When the [[Enumerate]] internal method of _O_ is called, the following steps are taken:</p>
       <emu-alg>
-        1. Return an Iterator object (<emu-xref href="#sec-iterator-interface"></emu-xref>) whose `next` method iterates over all the String-valued keys of enumerable properties of _O_. The Iterator object must inherit from %IteratorPrototype% (<emu-xref href="#sec-%iteratorprototype%-object"></emu-xref>). The mechanics and order of enumerating the properties is not specified but must conform to the rules specified below.
+        1. Return ? OrdinaryEnumerate(_O_).
```

Comment:
This entire clause will change once #367 is accepted. Perhaps undo any changes to this clause so rebase is easier. OrdinaryEnumerate is now basically EnumerateObjectProperties in #367 so from a layering perspective should work identically.